org.scala-ide.play2/src/org/scalaide/play2/templateeditor/BracketAutoEditStrategy.scala

```diff
@@ -0,0 +1,38 @@
+package org.scalaide.play2.templateeditor
+
+import org.eclipse.jface.preference.IPreferenceStore
+import org.eclipse.jface.text.DocumentCommand
+import org.eclipse.jface.text.IAutoEditStrategy
+import org.eclipse.jface.text.IDocument
+
+/**
+ * Copied from scala-ide.core since the class has been removed.
```

Comment:
If that was a public (non-internal class) you need to add it back. We can't have binary incompatible changes in the 4.0 series.